lib/ok.ex

```diff
@@ -150,32 +150,52 @@ defmodule OK do
   end
 
   @doc """
-  Combine multiple functions that may return an error.
-
-  `OK.with/1 is a strict version of the native elixir with special form.
-  It is stricter because it will only handle result tuples.
-  This however allows for more terse expressions.
+  Composes multiple functions similar to Elixir's native `with` construct.
+  
+  `OK.with/1` enables more terse and readable expressions however, eliminating
+  noise and regaining precious horizontal real estate in the process. It does 
+  this by extracting result tuples when using the `<-` operator.
+  
+  For example, in native Elixir, you could write a `with` statement as follows:
+  
+  ```elixir
+  with {:ok, a} <- safe_div(8,2),
```

Comment:
I think we should keeps all example code in the examples section to it can be tested by the doc tests. even if it is not using the library code